Sources/Utilities/KMZendeskChatHandler.swift

```diff
@@ -58,71 +77,131 @@ public class KMZendeskChatHandler {
+        sendChatTranscript()
+    }
+
+    
     public func getGroupId() -> String {
         return groupId
     }
+    public func updategroupId(_ groupId: String) {
+        self.groupId = groupId
+        self.updateLastMessageCreatedTime()
+    }
+   
     func authenticateUser() {
-        let chatAPIConfiguration = ChatAPIConfiguration()
-        chatAPIConfiguration.visitorInfo = VisitorInfo(name: ALUserDefaultsHandler.getDisplayName() ?? "", email: ALUserDefaultsHandler.getEmailId() ?? "", phoneNumber: "")
-        Chat.instance?.configuration = chatAPIConfiguration
-        connectToZendeskSocket()
+        let userHandler = ALUserDefaultsHandler.self
+        
+        if let externalId = userHandler.getUserId(),!externalId.isEmpty,
+           let name = userHandler.getDisplayName(),!name.isEmpty,
+           let email = userHandler.getEmailId(),!email.isEmpty {
+            authenticateUserWithJWT(name: name, email: email, externalId: externalId)
+        }
     }
-    var chatLogToken : ObservationToken?
     
-    func observeChatLogs() {
-        guard Chat.connectionProvider?.status == .connected else {
-            connectToZendeskSocket()
-            return
+    
+    
+    func authenticateUserWithJWT(name: String, email: String, externalId: String) {
+        let dictionary = ["name":name, "email": email, "externalId":externalId]
+
+        let postdata: Data? = try? JSONSerialization.data(withJSONObject: dictionary, options: [])
+        var theParamString: String? = nil
+        if let aPostdata = postdata {
+            theParamString = String(data: aPostdata, encoding: .utf8)
         }
         
-        let stateToken = Chat.chatProvider?.observeChatState { (state) in
-            // Handle logs, agent events, queue position changes and other events
+        guard let postURLRequest = ALRequestHandler.createPOSTRequest(withUrlString: "https://api.kommunicate.io/rest/ws/zendesk/jwt", paramString: theParamString) as NSMutableURLRequest? else { return }
+        let responseHandler = ALResponseHandler()
+        
+        responseHandler.authenticateAndProcessRequest(postURLRequest, andTag: "") {
+            (json, error) in
+            guard error == nil else {
+                return
+            }
+            guard let dict = json as? [String: Any], let data = dict["data"] as? [String: Any], let jwtKey = data["jwt"] as? String else {
+                return
+            }
+            self.jwtToken = jwtKey
+            // Connect to their server after authentication
+            self.connectToZendeskSocket()
+
         }
-        chatLogToken = stateToken
     }
     
+    var chatLogToken : ObservationToken?
+    
     func connectToZendeskSocket() {
-        Chat.connectionProvider?.connect()
+        guard let connectionProvider = Chat.connectionProvider else {
+            return
+        }
+        connectionProvider.connect()
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
Why delay? What if we missed connection logs?

Reply:
yes there is a chance of missing logs. Removed the delay